rust/src/live.rs

```diff
@@ -479,11 +466,34 @@ pub(crate) fn transaction_apply_live(
     // Success! Update the recorded state.
     state.commit = target_commit.to_string();
     state.inprogress = "".to_string();
-    write_live_state(&booted, &state)?;
+    write_live_state(&repo, &booted, &state)?;
 
     Ok(())
 }
 
+/// Writing a ref for the live-apply state can get out of sync
+/// if we upgrade.  This prunes the ref if the booted deployment
+/// doesn't have a live apply state in /run.
+pub(crate) fn applylive_sync_ref(
+    mut sysroot: Pin<&mut crate::ffi::OstreeSysroot>,
+) -> CxxResult<()> {
+    let sysroot = sysroot.gobj_wrap();
+    let repo = &sysroot.get_repo(gio::NONE_CANCELLABLE)?;
+    let booted = if let Some(b) = sysroot.get_booted_deployment() {
+        b
+    } else {
+        return Ok(());
+    };
+    if get_live_state(&repo, &booted)?.is_some() {
+        return Ok(());
+    }
+
+    // Set the live state to empty
+    let state = Default::default();
+    write_live_state(&repo, &booted, &state)?;
```

Comment:
In this case, we don't want it to touch the stamp file, no?

Reply:
Well spotted indeed!